test/regression/regression_test.cpp

```diff
@@ -0,0 +1,79 @@
+//
+// Created by igor on 07.03.18.
+//
+
+#include <gtest/gtest.h>
+
+#include "builders/protobuf/transaction.hpp"
+#include "cryptography/ed25519_sha3_impl/internal/ed25519_impl.hpp"
+#include "framework/integration_framework/integration_test_framework.hpp"
+
+constexpr auto kUser = "user@test";
+constexpr auto kAsset = "asset#domain";
+const auto kAdminOldKeypair = iroha::create_keypair();
+const shared_model::crypto::Keypair kAdminKeypair(
+    shared_model::crypto::PublicKey(kAdminOldKeypair.pubkey.to_string()),
+    shared_model::crypto::PrivateKey(kAdminOldKeypair.privkey.to_string()));
+
+/**
+ * @given ITF instance with Iroha
+ * @when existing ITF instance was not gracefully shutdown
+ * @then following ITF instantiation should not cause any errors
+ */
+TEST(RegressionTest, SequentialInitialization) {
+  auto tx = shared_model::proto::TransactionBuilder()
+                .createdTime(iroha::time::now())
+                .creatorAccountId(kUser)
+                .txCounter(1)
+                .addAssetQuantity(kUser, kAsset, "1.0")
+                .build()
+                .signAndAddSignature(
+                    shared_model::crypto::DefaultCryptoAlgorithmType::
+                        generateKeypair());
+
+  auto checkStatelessValid = [](auto &status) {
+    ASSERT_NO_THROW(
+        boost::get<shared_model::detail::
```

Comment:
You may use [SpecifiedVisitor](https://github.com/hyperledger/iroha/blob/develop/shared_model/interfaces/utils/specified_visitor.hpp#L28), like here:
https://github.com/hyperledger/iroha/blob/afd5bd1abed98524dd63662a6deb579ac8e2f87e/test/integration/acceptance/invalid_fields_test.cpp#L59-L61

Reply:
up